### DISCUSSION TITLE
Gcamp8s denoising?

### DISCUSSION THREAD

**Discussion by cawarwick** [flatironinstitute/CaImAn#1371] [General]
I wonder if there are some specific things that are necessary to change for the newest generation of gcamps? 
I have below an example of the raw fluorescence from a 2p Gcamp8s recording from a couple of example cells. These cells are being stimulated with 1, 2, 3, 4, 5 APs with a 50ms ISI and recorded at 114hz (x-axis is frames).
![image](https://github.com/flatironinstitute/CaImAn/assets/81972652/20ab3fc4-9087-47c7-ade5-78b01d277a0a)

When I process the data using a caiman pipeline I had made for gcamp6s (which worked fine with gcamp6s at 10hz), it seems to introduce a significant number of artifacts when checking the denoised traces (x axis is in seconds this time, but it's the same recording with 1, 2, 3, 4, 5 APs). 
![image](https://github.com/flatironinstitute/CaImAn/assets/81972652/c04dee19-9c66-4665-8f4e-668e0e20ca05)

If I overlap the approximately same cell there is definitely some weirdness happening. Here black is a manual ROI from ImageJ and just a plain DFF normalization and green is the roughly equivalent component from Caiman showing 1,2,3 APs
![image](https://github.com/flatironinstitute/CaImAn/assets/81972652/b2e536aa-4d2d-4c01-9395-852dfe9bc78a)

I'm not sure whether the denoising process is highly affected by decay_time but I've tried 1.0 and 0.3 and still saw similar problems, but I wasn't sure whether that variable gets used in the Denoising. Thanks!

As an addendum, here are some high S:N cells with a single AP in gcamp6s (blue) processed with caiman vs gcamp8s (red) processed with caiman. The Gcamp6s looks exactly how I'd expect, but the Gcamp8s introduces some weird blips that just aren't there in the raw fluorescence. 
![image](https://github.com/flatironinstitute/CaImAn/assets/81972652/88b0ca45-b6ba-4e00-99c8-0fdc51d27d96)

**Comment by kushalkolar** [comment]
This looks like a deconvolution issue. What are you setting for the order of the AR process?
What do you get if you disable deconvolution?

**Reply by cawarwick** [reply]
I was plotting the C not the S here so I assumed that it didn't matter? Does performing the decon also affect the C? I was using P=2 since the Gcamp8s was sampled at >100hz and while it's not a lot of points on the upwing, there were some so thought P=2 seemed ok. (EDIT for anyone else who came across this, C is the denoised AND deconvolved calcium so it is affected by the value of P)

![image](https://github.com/flatironinstitute/CaImAn/assets/81972652/497884f6-8bc5-48cb-aa2e-7c84dee31f05)

I will try running it with decon off (p=0) and see what it looks like, thanks

**Reply by cawarwick** [reply]
Definitely seems to be related to having the deconvolution turned on. I totally blanked on the fact that the estimates.C are the denoised AND deconvolved calcium traces. But yeah, it's definitely deconvolution related. I haven't messed with constrained_foopsi or constrained_oasisAR2 at all, but it sounds like that's what I will need to mess with to get the gcamp8 to work with the full pipeline? Any suggestions for what to change about it? Thanks for your help, greatly appreciated 

These are the estimates with P=0
![image](https://github.com/flatironinstitute/CaImAn/assets/81972652/8f219543-3dd4-4176-a1ed-872d80501c2b)

![image](https://github.com/flatironinstitute/CaImAn/assets/81972652/5ad64c89-8258-46da-ac58-e13e9c8aaf42)
